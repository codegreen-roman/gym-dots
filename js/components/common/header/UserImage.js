import React from 'react'
import glamorous from 'glamorous'
import { string } from 'prop-types'

const ImgForUser = glamorous.img({
    width: '3rem',
    borderRadius: '2rem'
})

export const UserImage = ({ image }) => (
    <ImgForUser src={image} />
)

UserImage.defaultProps = {
    image: ''
}

UserImage.propTypes = {
    image: string
}
