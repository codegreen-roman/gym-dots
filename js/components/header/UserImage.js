import React from 'react'
import { image } from 'faker'
import glamorous from 'glamorous'

const userPic = image.avatar()

const ImgForUser = glamorous.img({
    width: '3rem',
    borderRadius: '2rem'
})

export const UserImage = () => (
    <ImgForUser src={userPic} />
)
