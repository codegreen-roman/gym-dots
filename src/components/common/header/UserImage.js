import React from 'react'
import glamorous from 'glamorous'
import { string } from 'prop-types'

const ImgForUser = glamorous.img({
    width: '3rem',
    borderRadius: '2rem'
})

export const UserImage = ({ image }) => {

    if (image) return <ImgForUser src={image} alt='This is your avatar' />
    return null
}

UserImage.propTypes = {
    image: string
}
