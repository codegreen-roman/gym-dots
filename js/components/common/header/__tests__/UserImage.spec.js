import { matcher, serializer } from 'jest-glamor-react'

expect.extend(matcher)
expect.addSnapshotSerializer(serializer)

import React from 'react'
import { mount } from 'enzyme'
import { UserImage } from '../UserImage'

describe('UserImage component', () => {

    const setup = () => mount(<UserImage />)

    it('should render the image for the user', () => {
        const wrapper = setup()
        expect(wrapper).toMatchSnapshotWithGlamor()
    })

})
