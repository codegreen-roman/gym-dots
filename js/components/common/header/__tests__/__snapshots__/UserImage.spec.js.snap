// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserImage component should render the image for the user 1`] = `
<UserImage>
  <glamorous(img)
    src="https://s3.amazonaws.com/uifaces/faces/twitter/antjanus/128.jpg"
  >
    <img
      className="css-5sn766"
      src="https://s3.amazonaws.com/uifaces/faces/twitter/antjanus/128.jpg"
    />
  </glamorous(img)>
</UserImage>
`;
